staging/src/k8s.io/client-go/tools/clientcmd/validation.go

```diff
@@ -254,6 +257,21 @@ func validateClusterInfo(clusterName string, clusterInfo clientcmdapi.Cluster) [
 		} else {
 			defer clientCertCA.Close()
 		}
+		certificateAuthorityData, err := io.ReadAll(clientCertCA)
```

Comment:
be careful about making changes that require double reading the file in a single invocation… that breaks invocations that pipe data into single-use file descriptors (like `… <(command to generate a key or CA)`